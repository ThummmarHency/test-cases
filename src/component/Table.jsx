import React from "react";

const Table = () => {
  return (
    <table className="custome-border">
      <thead>
        <tr>
          <td>id</td>
          <td>names</td>
        </tr>
      </thead>
      <tbody>
          <tr>
              <td>001</td>
              <td>test</td>
          </tr>
      </tbody>
    </table>
  );
};

export default Table;
