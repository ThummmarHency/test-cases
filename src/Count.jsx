import React, { useState } from "react";
import Table from "./component/Table";

const Count = () => {
  const [count, setCount] = useState(0);
  const handleCount = () => {
    setCount(count + 1);
  };
  return (
    <>
      <div className={`count-${count}`}>{count}</div>
      <button onClick={handleCount}>Click</button>
      <Table/>

    </>
  );
};

export default Count;
