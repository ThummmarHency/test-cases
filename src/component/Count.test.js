import React from 'react'

import  { shallow } from "enzyme";
import Count from '../Count';
import Table from './Table';
import toJson from 'enzyme-to-json';

describe.skip("Count page testing",()=>{
    const countInstance=shallow(<Count/>)

    it("simulate check",()=>{
        expect(countInstance.find('.count-0').length).toEqual(1)
        countInstance.find('button').simulate('click')
        expect(countInstance.find('.count-1').length).toEqual(1)
    })
    it("dive",()=>{
        expect(countInstance.find(Table).dive().find('.custome-border').length).toBe(1)
    })
    it("snapShot tabel",()=>{
        const tableInstance=shallow(<Table/>)
        expect(toJson(tableInstance)).toMatchSnapshot()
    })
    it("snapShot table 1",()=>{
        const tableInstance=shallow(<Table/>)
        expect(toJson(tableInstance)).toMatchSnapshot()
    })
    it("snapShot table 2",()=>{
        const tableInstance=shallow(<Table/>)
        expect(toJson(tableInstance)).toMatchSnapshot()
    })
})