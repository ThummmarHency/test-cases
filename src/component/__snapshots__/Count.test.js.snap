// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Count page testing snapShot tabel 1`] = `
<table
  className="custome-border"
>
  <thead>
    <tr>
      <td>
        id
      </td>
      <td>
        names
      </td>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        001
      </td>
      <td>
        test
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Count page testing snapShot table 1 1`] = `
<table
  className="custome-border"
>
  <thead>
    <tr>
      <td>
        id
      </td>
      <td>
        names
      </td>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        001
      </td>
      <td>
        test
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Count page testing snapShot table 2 1`] = `
<table
  className="custome-border"
>
  <thead>
    <tr>
      <td>
        id
      </td>
      <td>
        names
      </td>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        001
      </td>
      <td>
        test
      </td>
    </tr>
  </tbody>
</table>
`;
